src/soter/openssl/soter_ec_key.c

```diff
@@ -114,33 +160,60 @@ soter_status_t soter_engine_specific_to_ec_pub_key(const soter_engine_specific_e
     EVP_PKEY* pkey = (EVP_PKEY*)engine_key;
     soter_status_t res;
     size_t output_length;
+#ifndef THEMIS_EXPERIMENTAL_OPENSSL_3_SUPPORT
     EC_KEY* ec;
     const EC_GROUP* group;
     const EC_POINT* Q;
     int curve;
+#else
+    char group_str[16];
+    const char* param_name;
+    size_t serialized_len;
+#endif
 
     if ((!key_length) || (EVP_PKEY_EC != EVP_PKEY_id(pkey))) {
         return SOTER_INVALID_PARAMETER;
     }
 
+#ifndef THEMIS_EXPERIMENTAL_OPENSSL_3_SUPPORT
     ec = EVP_PKEY_get1_EC_KEY((EVP_PKEY*)pkey);
     if (NULL == ec) {
         return SOTER_FAIL;
     }
 
+    // Curve identifier as EC_GROUP*
     group = EC_KEY_get0_group(ec);
     if (NULL == group) {
         res = SOTER_INVALID_PARAMETER;
         goto err;
     }
 
+    // Curve identifier as int (NID_X9_62_prime256v1, NID_secp384r1, NID_secp521r1 etc)
     curve = EC_GROUP_get_curve_name(group);
     if (!is_curve_supported(curve)) {
         res = SOTER_INVALID_PARAMETER;
         goto err;
     }
 
     output_length = ec_pub_key_size(curve, compressed);
+#else
+    // Curve identifier as string ("prime256v1", "secp384r1", "secp521r1" etc)
+    if (!EVP_PKEY_get_utf8_string_param(pkey,
+                                        OSSL_PKEY_PARAM_GROUP_NAME /* "group" */,
+                                        group_str,
+                                        sizeof(group_str),
+                                        NULL)) {
```

Comment:
I see you like string-identified properties. Take a look at [NodeJS crypto module](https://nodejs.org/api/crypto.html#class-cipher) :)

Reply:
WebCryptography is like that as well.

Well, anyway, I get the appeal of not having to meticulously maintain a global list of numeric constants and having to invent funny ways to namespace them so that it is possible for plugins to register their own cryptosystems without stepping on each others toes.